manifests/config.pp

```diff
@@ -73,6 +73,15 @@
       'certname': value => $client_certname;
     }
   }
+  if $hostprivkey {
+    puppet::config::main {
+      'hostprivkey': value => $hostprivkey;
+    }
+  } else {
+    puppet::config::main {
+      'hostprivkey': value => '$privatekeydir/$certname.pem { mode = 640 }';
+    }
+  }
```

Comment:
if we set `$puppet::params::hostprivkey` default (like below), can remove this I think